### PR TITLE
Include sound samples in Dirt output

### PR DESCRIPTION
**Pull request by anderspapitto**
also add self as maintainer

### CODE REVIEW

File: pkgs/applications/audio/dirt/default.nix [hunk 1]
```diff
@@ -12,10 +12,14 @@ stdenv.mkDerivation rec {
   configurePhase = ''
     export DESTDIR=$out
   '';
+  postInstall = ''
+    cp -r samples $out/samples
```
Comment (vcunat) [inline pkgs/applications/audio/dirt/default.nix]: Are you sure that's the intended location? I'd expect it somewhat deeper, e.g. somewhere under ``$out/share/``.

File: lib/maintainers.nix [hunk 2]
```diff
@@ -9,6 +9,7 @@
   aforemny = "Alexander Foremny <alexanderforemny@googlemail.com>";
   aflatter = "Alexander Flatter <flatter@fastmail.fm>";
   aherrmann = "Andreas Herrmann <andreash87@gmx.ch>";
+  ajp = "Anders Papitto <anderspapitto@gmail.com>";
```
Comment (vcunat) [inline lib/maintainers.nix]: It's very helpful when the maintainer attrname is the same as github username. One can then easily ping the maintainer by @anderspapitto.

**Comment by anderspapitto** [comment]
addressed comments

**Comment by copumpkin** [comment]
This seems like a perfect example of something that could use multiple outputs. Are the samples big?

**Comment by anderspapitto** [comment]
@copumpkin yes, much larger than the binary itself. Is multiple outputs an existing feature which you would like me to use?

**Comment by copumpkin** [comment]
Yeah, I think it could be good. This allows the garbage collector to collect the samples independently from the main package. I'm not sure how well documented the multi-output stuff is yet, but there's some info in https://gist.github.com/vcunat/6139ee17ae1dec684fd3#comment-1318994.

From your angle, it should be basically as simple as saying 

```nix
mkDerivation {
  outputs = [ "out" "samples" ]; # Or whatever names make sense

  installPhase = ''
    mkdir -p $samples/share/dirt/samples
    cp someSample $samples/share/dirt/samples

    # Do regular stuff with #out
  '';
}
```

**Comment by anderspapitto** [comment]
Now using multiple inputs. Also applied a small patch to tell it where to find the samples (is it ok for one of the outputs to reference the other like this)? Without the patch the program is looking in "./samples", so it was necessary to do some hacky 'cd-into-the-nix/store-next-to-the-binary' to run. With multiple outputs that hacky approach wouldn't work at all any more since there's no way to know where the samples are.

**Comment by vcunat** [comment]
IMHO the main point of using splitting (either via multiple-output derviation or separately-built derivations) is to cover cases where often one doesn't need/want all that is produced, e.g. if it's meaningful to install the program without samples or samples without program.

If you split+patch binaries like this, you *always* get samples with the program (like before splitting), but now you can use samples without the program.

**Comment by copumpkin** [comment]
Yeah, I didn't realize there was a strong dependency between the two. That eliminates the opportunity to independently GC the expensive one, which was most of the point. Sorry for leading you down the wrong path!

**Comment by anderspapitto** [comment]
@copumpkin no problem, now I know about multiple outputs.

It looks like for this particular package, single output is probably the way to go. I've re-updated the PR and tested that it still works

**Comment by vcunat** [comment]
OK. I was about to point out the missing version of dirt, but that got fixed in the meantime.

**Pull request closed without merging**